Allow the complete automatic provisioning of default options during the install script

szaimen opened the pull request:
As the title says

This PR fixes #1523

szaimen commented:
What looks better to you:
`PROVISIONING=1 bash nextcloud_install_production.sh`
or
`bash nextcloud_install_production.sh --provisioning` ?
Both is possible...

szaimen commented:
We should probably change that command here too: https://github.com/nextcloud/vm/blob/a4eb6a6ac44059c5a800d0b6f103ac15d1c6c1f2/vagrant/install.sh#L13

enoch85 commented:
So this removes all questions if `--provisioning` is set?

enoch85 commented:
> bash nextcloud_install_production.sh --provisioning ?
Both is possible...

:+1:

szaimen commented:
> So this removes all questions if `--provisioning` is set?

yes 👍

szaimen commented:
I've added it to the vagrant file. Also I added a short `-p` flag to do the same.

btw: I've already tested this successfuly :D

The only remaining question is this:
> BTW: I think the cloning is unnecessary (we only need the install_script, no?) :
> Or would wget and curl not work on vagrant?
> https://github.com/nextcloud/vm/blob/a4eb6a6ac44059c5a800d0b6f103ac15d1c6c1f2/vagrant/install.sh#L9

enoch85 commented:
Is this done?

szaimen commented:
Yes, I only need an answer for this

> BTW: I think the cloning is unnecessary (we only need the install_script, no?) :
> Or would wget and curl not work on vagrant?
> [vm/vagrant/install.sh](https://github.com/nextcloud/vm/blob/a4eb6a6ac44059c5a800d0b6f103ac15d1c6c1f2/vagrant/install.sh#L9)
> 
> Line 9 in [a4eb6a6](/nextcloud/vm/commit/a4eb6a6ac44059c5a800d0b6f103ac15d1c6c1f2)

szaimen commented:
@enoch85 I think this is ready now.

enoch85 commented on vagrant/install.sh:
```diff
@@ -4,11 +4,10 @@
 
-cd vm || exit && print_text_in_color "$IRed" "Could not cd into the 'vm' folder."
-
-yes no | sudo bash nextcloud_install_production.sh
```
The vm folder needs to be local. Hence the code for that.

szaimen replied:
I doubt that. We have nothing implemented to use the local scripts.
I am pretty sure that it just runs the script like everywhere else. (Loads the scripts during nextcloud_iinstall_production.sh directly from github)

enoch85 replied:
Well, have you actually tested it? 

I know for a fact the the current implementation works, so of you don't want to test the new one, please revert. :)

szaimen replied:
Okay, I'll do 👍

szaimen replied:
done

szaimen commented:
I think this is ready
